plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveSplitSource.java

```diff
@@ -362,6 +367,9 @@ else if (maxSplitBytes * 2 >= remainingBlockBytes) {
                     splitBytes = internalSplit.getEnd() - internalSplit.getStart();
                 }
 
+                long currentSplitNumber = numberOfProcessedSplits.getAndIncrement();
+                checkState(currentSplitNumber < MAX_NUMBER_OF_SPLITS, "Number of splits is higher than maximum possible number of splits");
+                OptionalLong initialRowId = OptionalLong.of(currentSplitNumber * MAX_NUMBER_OF_ROWS_PER_SPLIT);
```

Comment:
I would prefer `currentSplitNumber << 42`.

Also please extract `42` as `PER_SPLIT_ROW_ID_BITS` and `64 - PER_SPLIT_ROW_ID_BITS` as `SPLIT_ID_BITS`

Reply:
ok